### PR TITLE
Apply nav menu classes only if `theme_location` parameter is populated

### PR DESCRIPTION
**Pull request by crstauf**
## What does this do/fix?

Improves the classes added to nav menu elements by supporting the `menu` parameter (of `wp_nav_menu()`). See #880.

## Tests

Does this have tests?

- [ ] Yes
- [ ] No, this doesn't need tests because...
- [x] No, I need help figuring out how to write the tests.

After creating two menus (`sample` and `primary`) and assigning one of them to the `primary` theme location, added the following code to an mu-plugin to easily test the proper output:

```php
<?php declare(strict_types=1);

use Tribe\Project\Nav_Menus\Menu;

add_action( 'wp_body_open', function() {

	// Test theme_location parameter.
	echo Menu::menu( [
		'theme_location' => 'primary',
		'container_class' => 'test-1 location-primary',
	] );

	// Test menu parameter.
	echo Menu::menu( [
		'menu' => 'sample',
		'container_class' => 'test-2 menu-sample',
	] );

	// Test menu and theme_location parameter.
	echo Menu::menu( [
		'menu' => 'sample',
		'theme_location' => 'primary',
		'container_class' => 'test-3 menu-sample',
	] );

	// Test invalid menu and valid theme_location parameter.
	echo Menu::menu( [
		'menu' => 'foo',
		'theme_location' => 'primary',
		'container_class' => 'test-4 location-primary',
	] );

	// Test valid menu and invalid theme_location parameter.
	echo Menu::menu( [
		'menu' => 'sample',
		'theme_location' => 'bar',
		'container_class' => 'test-5 menu-sample',
	] );

	exit;
} );
```

### CODE REVIEW

File: wp-content/plugins/core/src/Nav_Menus/Nav_Attribute_Filters.php
Comment (defunctl) [inline wp-content/plugins/core/src/Nav_Menus/Nav_Attribute_Filters.php:29]: This should fail PHPCS

File: wp-content/plugins/core/src/Nav_Menus/Nav_Attribute_Filters.php
Comment (defunctl) [inline wp-content/plugins/core/src/Nav_Menus/Nav_Attribute_Filters.php:32]: So should this :)

File: wp-content/plugins/core/src/Nav_Menus/Nav_Attribute_Filters.php
Comment (defunctl) [inline wp-content/plugins/core/src/Nav_Menus/Nav_Attribute_Filters.php:35]:
```suggestion
		// Get slug of menu if it exists.
```

File: wp-content/plugins/core/src/Nav_Menus/Nav_Attribute_Filters.php
Comment (defunctl) [inline wp-content/plugins/core/src/Nav_Menus/Nav_Attribute_Filters.php:36]: `wp_get_nav_menu_object()` is filterable, I'd check directly that `$menu->slug` itself exists, and as a suggestion you could combine that with the initial define which improves readability. 

At some point scrolling through this large code block you have to back up and check if things have been initialized or not.

e.g.

```php
$menu_slug = ( $menu->slug  ?? '' ) ? $menu->slug : '';
```

**Review by defunctl — CHANGES_REQUESTED** [review]
This would actually be a really great use case for https://github.com/moderntribe/square1-pipeline. 

Rather than a large method with lots of "if" statements, you can move the logic into defined steps/stages and skip to the next step if a condition or two isn't met. 

That would also make each step object easily testable with integration tests, where right now this would be very difficult to write tests for, given the number of conditions in a single method.

**Review by dpellenwood — COMMENTED** [review]
@crstauf thank you for taking a stab at this. However, it actually leads to more issues than it solves. I'd like to chat with you in person about it before we proceed with another approach.

File: wp-content/plugins/core/src/Nav_Menus/Nav_Attribute_Filters.php
Comment (crstauf) [reply wp-content/plugins/core/src/Nav_Menus/Nav_Attribute_Filters.php:36]: > check directly that $menu->slug itself exists

👍 

> At some point scrolling through this large code block you have to back up and check if things have been initialized or not.

I very rarely use ternary operators, as it decreases readability, especially for junior devs. Remembering the variables were initialized or scrolling up a little bit to confirm is much easier. ✌️

**Review by dpellenwood — CHANGES_REQUESTED** [review]
Thanks again for chatting through this with me, @crstauf . Per our conversation, please take the approach of checking for a theme location and if there is not one provided, existing early from the class filters. We don't (ever) want to create CSS classes for menus & items based on the menu ID/slug because those are specific to the data of the menu, not generic to a menu's template/usage/style.

**Comment by nickpelton** [comment]
Thanks everyone for popping on and chatting this through, good discussion leads to better tools for us all.

**Review by dpellenwood — APPROVED** [review]
:rocket: Thanks again for being patient while we all hashed this out together!

**Review by nickpelton — APPROVED** [review]

**Comment by defunctl** [comment]
@crstauf don't forget a changelog update, please

**Comment by crstauf** [comment]
@defunctl Waiting for your approval.

**Review by defunctl — APPROVED** [review]
🚀

**Pull request merged**
